### ISSUE TITLE
Dynamic output in health checks cause abnormal activity

### ISSUE THREAD

**Issue by tiwilliam** [hashicorp/consul#1069]
This issue was originally brought up on the mailing list by Nicholas Capo.

It seems like health checks with dynamic output cause abnormal activity in the cluster. We have a +80 instance cluster with +120 health checks. We switched all these health checks from dynamic output (HTTP Date header) to static and noticed VERY big difference in server load, commit latency, last contact latency, heartbeat latency, RPC requests and more.

Consul configuration: https://gist.github.com/tiwilliam/f05a66c52f85b000677f

Health check output before:

```
HTTP/1.1 200 OK
Date: Mon, 29 Jun 2015 19:04:40 GMT
Content-Type: application/json
Content-Length: 17
Server: gunicorn/19.3.0

```

Health check output after:

```
{"status": "ok"}
```

Graphs from when making the switch:

![screen shot 2015-06-29 at 11 30 31](https://cloud.githubusercontent.com/assets/190349/8415984/d86713b4-1ea2-11e5-8adb-4c23b02f4c68.png)

![screen shot 2015-06-29 at 11 30 50](https://cloud.githubusercontent.com/assets/190349/8415986/da58e256-1ea2-11e5-8c07-91dae91a57ec.png)

**Comment by tiwilliam** [comment]
Here's a redacted copy of one agent before the switch:

**Comment by joshuaspence** [comment]
I also ran into this issue. This should be documented at the very least.

**Comment by slackpad** [comment]
This is related to https://github.com/hashicorp/consul/issues/1057 - it would be good to sync back periodically but not spam the servers with updates, especially if the service is healthy.

**Comment by slackpad** [comment]
Some basic rate limiting was added a while back, and I think we can close this against https://github.com/hashicorp/consul/issues/1982, which provides a good mechanism to automatically manage this. If we allow frequent updates for a while after a state change, we can throttle non-state-change updates even further without impacting the UX too much. We also did some work for Consul 0.8.x that avoids unnecessary updates to the service and/or node records when just a check changes, so this can cut the writes down to 1/3.
